RetireSimple.Tests/DomainModel/Sqlite/SqliteModelTest.cs

```diff
@@ -1,109 +1,155 @@
+            investment.StockTicker = "TST";
+
+            var investment2 = new StockInvestment("testAnalysis2");
+            investment2.StockPrice = 200;
+            investment2.StockQuantity = 20;
+            investment2.StockTicker = "TEST";
+
+            var options = new Dictionary<string, string>();
+
+
+            context.Investments.Add(investment);
+            context.Investments.Add(investment2);
+            context.SaveChanges();
+            context.InvestmentModels.Add(investment2.InvokeAnalysis(options));
+            context.InvestmentModels.Add(investment.InvokeAnalysis(options));
+            context.SaveChanges();
+
+            context.InvestmentModels.Should().HaveCount(2);
+        }
+
+        [Fact]
+        public void TestInvestmentModelFKConstraintonInvestment() {
+            var investment = new StockInvestment("testAnalysis");
+            investment.StockPrice = 100;
+            investment.StockQuantity = 10;
+            investment.StockTicker = "TST";
+
+            var investment2 = new StockInvestment("testAnalysis2");
+            investment2.StockPrice = 200;
+            investment2.StockQuantity = 20;
+            investment2.StockTicker = "TEST";
+
+            var options = new Dictionary<string, string>();
+
+            context.Investments.Add(investment);
+            context.Investments.Add(investment2);
+            context.SaveChanges();
+            context.InvestmentModels.Add(investment2.InvokeAnalysis(options));
+            context.InvestmentModels.Add(investment.InvokeAnalysis(options));
+            context.SaveChanges();
+
+            Action act = () => {
+                context.Investments.Remove(investment);
+            };
+
+            act.Should().Throw<InvalidOperationException>();
+        }
+
+        [Fact]
+        public void TestPortfolioAdd() {
+            var portfolio = new Portfolio();
+            var portfolio2 = new Portfolio();
+            var profile = new Profile();
+
+            context.Profiles.Add(profile);
+            context.Profiles.First(p => p.ProfileId == 1).Portfolios.Add(portfolio);
+            context.Profiles.First(p => p.ProfileId == 1).Portfolios.Add(portfolio2);
+
+
+            context.SaveChanges();
+            context.Portfolios.Should().HaveCount(2);
+        }
+
+        [Fact]
+        public void TestProfileAdd()
+        {
+            var profile = new Profile();
+            var profile2 = new Profile();
+
+            context.Profiles.Add(profile);
+            context.Profiles.Add(profile2);
+
+            context.SaveChanges();
+            context.Profiles.Should().HaveCount(2);
+        }
+
+        [Fact]
+        public void TestPortfolioFKConstraint()
+        {
+            var portfolio = new Portfolio();
+            var profile = new Profile();
+
+            context.Profiles.Add(profile);
+            context.Profiles.First(p => p.ProfileId == 1).Portfolios.Add(portfolio);

```

Comment:
Forgot to note that in order for this search to work ,the database needs to actually add the object, so this would fix that error
```suggestion
            context.SaveChanges();
            context.Profiles.First(p => p.ProfileId == 1).Portfolios.Add(portfolio);

```